static_frame/core/util.py

```diff
@@ -3186,5 +3187,91 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
+                # [139999659995264, 139999659995264, 139999659995264, 139999659995264, 139999659995264]
+                # >>> c
+                # [139999659995264, 139999658764368, 139999659995264, 139999658764368, 139999659995264]
+                # >>> d
+                # [array([None, None, 0], dtype=object), array([None, None, 0], dtype=object), array([None, None, 1], dtype=object), array([None, None, 1], dtype=object), array([None, None, 2], dtype=object)]
+                # >>> e
+                # [139999659995264, 139999658764368, 139999658765088, 139997430769344, 139997430769104]
+                yield from [el for el in obj]
+            elif (
+                isinstance(obj, abc.Sequence) # tuple, list
+                or isinstance(obj, abc.Set) # set, frozenset
+            ):
+                yield from obj
+            elif isinstance(obj, dict):
+                yield from chain.from_iterable(obj.items())
+            else:
+                # The full size of the object is included in its getsizeof call
+                # e.g. FrozenAutoMap, integer numpy arrays, int, float, etc.
+                pass
+
+    @staticmethod
+    def _sizable_slot_attrs(obj: tp.Any) -> tp.Iterable[tp.Any]:
+        '''
+        Generates an iterable of the values of all slot-based attributes in an object, including the slots
+        contained in the object's parent classes based on the MRO
+        '''
+        # NOTE: This does NOT support 'single-string' slots (i.e. __slots__ = 'foo')
+        slots = frozenset().union(*(cls.__slots__ for cls in obj.__class__.__mro__ if hasattr(cls, '__slots__')))
+        attrs = (getattr(obj, slot) for slot in slots if slot != '__weakref__' and hasattr(obj, slot))
+        return attrs
+
+    @staticmethod
+    def _nested_sizable_elements(obj: tp.Any, *, seen: tp.Set[int]) -> tp.Iterable[tp.Any]:
+        '''
+        Generates an iterable of all objects the parent object has references to, including nested references.
+        This function considers both the iterable unsized children (based on _unsized_children) and the sizable
+        attributes listed in its slots. The resulting generator is in pre-order and includes the parent object
+        at the end.
+        '''
+        if id(obj) in seen:
+            return
+        seen.add(id(obj))
+
+        for el in MemoryMeasurements._unsized_children(obj):
+            yield from MemoryMeasurements._nested_sizable_elements(el, seen=seen)
+        for el in MemoryMeasurements._sizable_slot_attrs(obj):
+            yield from MemoryMeasurements._nested_sizable_elements(el, seen=seen)
+
+        yield obj
+
+def getsizeof_total(obj: tp.Any, *, seen: tp.Union[None, tp.Set[tp.Any]] = None) -> int:
+    '''
+    Returns the total size of the object and its references, including nested refrences
+    '''
+    seen = set() if seen is None else seen
+    total = sum(getsizeof(el) for el in MemoryMeasurements._nested_sizable_elements(obj, seen=seen))
```

Comment:
Lets make the public entry point explicit:
```suggestion
    total = sum(getsizeof(el) for el in MemoryMeasurements.nested_sizable_elements(obj, seen=seen))
```